packages/matrix-client-server/src/user/account_data/index.ts

```diff
@@ -0,0 +1,98 @@
+import type MatrixClientServer from '../..'
+import {
+  jsonContent,
+  validateParameters,
+  errMsg,
+  type expressAppHandler,
+  send
+} from '@twake/utils'
+
+interface parameters {
+  userId: string
+  type: string
+}
+
+interface PutRequestBody {
+  content: string
+}
+
+const schema = {
+  content: true
+}
+
+const mxidRe = /^@[0-9a-zA-Z._=-]+:[0-9a-zA-Z.-]+$/
+// TODO : Handle error 403 where the user isn't allowed to see this user's account data
+// TODO : Handle error 405 where the type of account data is controlled by the server and cannot be modified by the client
+
+const accountDataType = (
+  clientServer: MatrixClientServer
+): expressAppHandler => {
+  return (req, res) => {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-expect-error
+    const prms: parameters = req.params as parameters
+    if (!mxidRe.test(prms.userId)) {
+      send(res, 400, errMsg('invalidParam', 'invalid Matrix user ID'))
+      return
+    }
+    clientServer.authenticate(req, res, (data, id) => {
+      if (req.method === 'GET') {
```

Comment:
Method is too long, too much indents.
That's a code smell.
A easy workaround is to extrat handling for PUT / GET into speparated metods.
It would make it easier to follow the code flows and give name / explicit intermediary operations.

Reply:
Will do that right away